Source/WTF/wtf/glib/RunLoopGLib.cpp

```diff
@@ -282,6 +347,17 @@ void RunLoop::TimerBase::updateReadyTime()
 
 void RunLoop::TimerBase::start(Seconds interval, bool repeat)
 {
+#if HAVE(TIMERFD)
+    // Create the timerfd here so that it's created as late as possible. Some
+    // timers are created but may never be triggered.
+    auto& runLoopSource = *reinterpret_cast<RunLoopSource*>(m_source.get());
+    if (m_interval && runLoopSource.timerFd < 0) {
+        runLoopSource.timerFd = timerfd_create(CLOCK_MONOTONIC, TFD_NONBLOCK | TFD_CLOEXEC);
+        if (runLoopSource.timerFd > -1)
+            g_source_add_unix_fd(m_source.get(), runLoopSource.timerFd, G_IO_IN);
```

Comment:
I think we might want to `LOG_ERROR()` here in case creating the timerfd failed, as it could be useful for debugging. If we are running out of file descriptors at least we would know what's going on instead of silently falling back to “less good” timing.


```suggestion
        if (runLoopSource.timerFd > -1) [[likely]]
            g_source_add_unix_fd(m_source.get(), runLoopSource.timerFd, G_IO_IN);
        else
            LOG_ERROR("Could not create timerfd: %s", safeStrerror(errno).data());
```